src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let resultString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let item = this[i];
+
+      if (this[i] === undefined || this[i] === null) {
+        item = '';
+      }
+
+      if (i === this.length - 1) {
+        resultString += `${item}`;
+      } else {
+        resultString += `${item}${separator}`;
+      }
```

Comment:
Just use `if` condition, without `else`. If you take the previous step, you can only use `if` condition here